### ISSUE TITLE
Producing at one blocked building lets blocking actors at other buildings nudge

### ISSUE THREAD

**Issue by abcdefg30** [OpenRA/OpenRA#10055]
I think only one of them should nudge here (although it can't hurt to move a the other blocker away, so not sure if we consider this a bug).
![nudgeexit](https://cloud.githubusercontent.com/assets/7704140/11319988/c90d73f8-908b-11e5-9d6c-99f343dadabd.gif)
![nudgeexit2](https://cloud.githubusercontent.com/assets/7704140/11320001/2fd010f0-908c-11e5-86fa-2fffb5a1a511.gif)

**Comment by pchote** [comment]
In principle this should be as simple as looping over all the structures to find a free one before doing any nudging.  In practice primary buildings make this tricky (should we nudge the unit out of the way of the primary, or build at another one?).

**Comment by abmyii** [comment]
Does this still occur @abcdefg30? I don't have the mod you were testing with.

**Comment by abcdefg30** [comment]
Yeah it still does. It should be present in the other mods too.

**Comment by pchote** [comment]
#17141 will fix this for TS, but the new logic there has been WONTFIXED for the other mods.

**Comment by abmyii** [comment]
Why not? Though it doesn't seem to occur in RA.

**Comment by deleted-user-1** [comment]
>Though it doesn't seem to occur in RA.

It does.

<details>

![production nudge](https://user-images.githubusercontent.com/59416324/76169640-93b15c00-617a-11ea-97a0-d94992f4cceb.gif)

**Comment by abmyii** [comment]
> > Though it doesn't seem to occur in RA.
> 
> It does.
> 
> ![production nudge](https://user-images.githubusercontent.com/59416324/76169640-93b15c00-617a-11ea-97a0-d94992f4cceb.gif)

Correct. Can't recall what I meant, but I made an issue for that as well (https://github.com/OpenRA/OpenRA/issues/17404).
